cloudevents/sdk/http_events.py

```diff
@@ -0,0 +1,113 @@
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import copy
+
+import json
+import typing
+
+from cloudevents.sdk import marshaller
+
+from cloudevents.sdk.event import base
+from cloudevents.sdk.event import v1
+
+
+class CloudEvent(base.BaseEvent):
+    """
+    Python-friendly cloudevent class supporting v1 events
+    Currently only supports binary content mode CloudEvents
+    """
+
+    def __init__(
+            self,
+            headers: dict,
+            data: any,
+            f: typing.Callable = lambda x: x
```

Comment:
Is `f` an idiomatic name here in python? I'm not sure.
Can we name this `callback`?